libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -47,15 +47,19 @@ class MapboxOffboardRouter(
                 response: Response<DirectionsResponse>
             ) {
                 val routes = response.body()?.routes()
-                if (response.isSuccessful && !routes.isNullOrEmpty()) {
-                    callback.onResponse(routes)
-                } else {
-                    callback.onFailure(NavigationException(ERROR_FETCHING_ROUTE))
+                when {
+                    call.isCanceled -> callback.onCanceled()
+                    response.isSuccessful && !routes.isNullOrEmpty() -> callback.onResponse(routes)
+                    else -> callback.onFailure(NavigationException(ERROR_FETCHING_ROUTE))
                 }
             }
 
             override fun onFailure(call: Call<DirectionsResponse>, t: Throwable) {
-                callback.onFailure(t)
+                if (call.isCanceled) {
+                    callback.onCanceled()
+                } else {
+                    callback.onFailure(t)
+                }
             }
         })
     }
```

Comment:
This change is not included in this PR, but I think we should add the following:

```
override fun cancel() {
        mapboxDirections?.cancelCall()
        mapboxDirections = null
}
```

Whenever `cancel()` is called, we should somehow notify the `directionSession` to stop the faster route fetcher timer. Thoughts?

Reply:
We're already doing that - whenever routes are empty, we're stopping the timer:
https://github.com/mapbox/mapbox-navigation-android/pull/2434/files#diff-009d81d6b9a5e6e4e15aa02120bf4aa1R29-R32
I believe this covers all of the cases.